android/src/ti/map/PolygonProxy.java

```diff
@@ -0,0 +1,302 @@
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.common.AsyncResult;
+import org.appcelerator.kroll.common.TiMessenger;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.TiContext;
+import org.appcelerator.titanium.util.TiConvert;
+
+import ti.map.Shape.IShape;
+import android.os.Message;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Polygon;
+import com.google.android.gms.maps.model.PolygonOptions;
+
+@Kroll.proxy(name = "Polygon", creatableInModule=MapModule.class, propertyAccessors = {
+
+MapModule.PROPERTY_FILL_COLOR, MapModule.PROPERTY_STROKE_COLOR,
+		MapModule.PROPERTY_STROKE_WIDTH, MapModule.PROPERTY_ZINDEX,
+		MapModule.PROPERTY_POINTS, PolygonProxy.PROPERTY_HOLES,
+
+})
+public class PolygonProxy extends KrollProxy implements IShape {
+
+	private PolygonOptions options;
+	private Polygon polygon;
+
+	private static final int MSG_FIRST_ID = KrollProxy.MSG_LAST_ID + 1;
+
+	private static final int MSG_SET_POINTS = MSG_FIRST_ID + 699;
+	private static final int MSG_SET_FILL_COLOR = MSG_FIRST_ID + 700;
+	private static final int MSG_SET_STROKE_COLOR = MSG_FIRST_ID + 701;
+	private static final int MSG_SET_STROKE_WIDTH = MSG_FIRST_ID + 702;
+	private static final int MSG_SET_ZINDEX = MSG_FIRST_ID + 703;
+	private static final int MSG_SET_HOLES = MSG_FIRST_ID + 704;
+
+	public static final String PROPERTY_HOLES = "holes";
+
+	public PolygonProxy() {
+		super();
+	}
+
+	public PolygonProxy(TiContext tiContext) {
+		this();
+	}
+
+	@Override
+	public boolean handleMessage(Message msg) {
+
+		// MSG_SET_POINTS
+		// MSG_SET_STROKE_WIDTH
+		// MSG_SET_STROKE_COLOR
+		// MSG_SET_FILL_COLOR
+		// MSG_SET_ZINDEX
+		AsyncResult result = null;
+		switch (msg.what) {
+		case MSG_SET_POINTS: {
+			result = (AsyncResult) msg.obj;
+			polygon.setPoints(processPoints(result.getArg(), true));
+			result.setResult(null);
+			return true;
+		}
+		case MSG_SET_HOLES: {
+			result = (AsyncResult) msg.obj;
+			polygon.setHoles(processHoles(result.getArg(), true));
+			result.setResult(null);
+			return true;
+		}
+		case MSG_SET_FILL_COLOR: {
+			result = (AsyncResult) msg.obj;
+			polygon.setFillColor((Integer) result.getArg());
+			result.setResult(null);
+			return true;
+		}
+		case MSG_SET_STROKE_WIDTH: {
+			result = (AsyncResult) msg.obj;
+			polygon.setStrokeWidth((Float) result.getArg());
+			result.setResult(null);
+			return true;
+		}
+		case MSG_SET_STROKE_COLOR: {
+			result = (AsyncResult) msg.obj;
+			polygon.setStrokeColor((Integer) result.getArg());
+			result.setResult(null);
+			return true;
+		}
+		case MSG_SET_ZINDEX: {
+			result = (AsyncResult) msg.obj;
+			polygon.setZIndex((Float) result.getArg());
+			result.setResult(null);
+			return true;
+		}
+		default: {
+			return super.handleMessage(msg);
+		}
+		}
+	}
+
+	public void processOptions() {
+
+		options = new PolygonOptions();
+		String op;
```

Comment:
Is the String op needed? Would it be possible to do the processOptions() method without it?